External source

alex-mo opened the pull request.

alex-mo committed:
Update JS setup

alex-mo committed:
Update transpiled pgbar.js

alex-mo committed:
Allow overriding the polling URL

alex-mo committed:
Add option to define JSON path to value

alex-mo committed:
Add some basic validations for the external source

alex-mo committed:
Compile with -b flag

PgbarItem was available globally and this should still be the case in
order to maintain backward compatibility.

alex-mo committed:
Fix typo

alex-mo committed:
Allow autostart to be set via the UI

This was hardcoded to TRUE.

alex-mo committed:
Provide external_url information to the JS

alex-mo committed:
Do not animateInitially() for external URLs

alex-mo committed:
Update targets dynamically

torotil commented on pgbar.coffee:
```diff
@@ -52,11 +54,26 @@ class PgbarItem
-      @animate(to_abs) if to_abs != @current
+      best_target = @selectTarget(to_abs)
+      if best_target != @settings.target
+        @settings.target = best_target
```
I don’t think it’s a good idea to use the settings to store a value. I would change it so that `@selectTarget()` is directly called from within `@animate()`.

alex-mo replied:
Changed.

torotil reviewed (commented):
1 suggested change. Otherwise it looks good!

alex-mo committed:
Update transpiled pgbar.js

alex-mo committed:
Make needed count updateable

alex-mo committed:
Set default values for the item

This fixes some noise because PHP will not trigger notices about
undefined indexes anymore.

alex-mo committed:
The span.pgbar-needed should not be a block

torotil commented on fields.inc.php:
```diff
@@ -90,8 +90,12 @@ function _pgbar_add_item_defaults(array $item) {
       ],
       'display' => [
         'template' => '',
+        'autostart' => FALSE,
```
Will this change the behaviour for existing field items without this value?

alex-mo replied:
Ah, good catch. `TRUE` should be the default I guess... :sweat_smile:
